src/libsyntax/diagnostic_list.rs

```diff
@@ -363,6 +363,29 @@ and likely to change in the future.
 
 "##,
 
+E0704: r##"
+This error indicates that a incorrect visibility restriction was specified.
+
+Example of erroneous code:
+
+```compile_fail,E0704
+mod foo {
+    pub(foo) struct Bar {
+        x: i32
+    }
+}
+```
+
+To make struct `Bar` only visible in module `foo` the `in` keyword should be used:
+```
+mod foo {
+    pub(in foo) struct Bar {
+        x: i32
+    }
+}
+```
```

Comment:
Could you extend a little bit with [relevant text from the book, or even a link to it](https://doc.rust-lang.org/reference/visibility-and-privacy.html#pubin-path-pubcrate-pubsuper-and-pubself)?


> In addition to public and private, Rust allows users to declare an item as visible within a given scope. The rules for pub restrictions are as follows:
> 
> * `pub(in path)` makes an item visible within the provided path. path must be a parent module of the item whose visibility is being declared.
> * `pub(crate)` makes an item visible within the current crate.
> * `pub(super)` makes an item visible to the parent module. This is equivalent to pub(in super).
> * `pub(self)` makes an item visible to the current module. This is equivalent to pub(in self).

Reply:
@estebank Sorry for the delay a bit! I added a line referencing the documentation! I thought it's best to not duplicate documentation too much so I hope it is good like this! I am not sure which style I should have used since the output is shown in a terminal. I was wondering if people can actually see the links. Let me know if I need to change something else!

Reply:
Good job on using a link. That should be good. I'm of the thought that we should provide enough information inline in the current place (page/error/etc.), with a way to find more info (easily googleable phrase or even better a link like in this case).